view/src/bin/pviewd.rs

```diff
@@ -1,15 +1,32 @@
 #![allow(clippy::clone_on_copy)]
 use anyhow::{Context, Result};
 use camino::Utf8PathBuf;
-use clap::{Parser, Subcommand};
+use clap::Parser;
+use penumbra_crypto::keys::{SeedPhrase, SpendKey};
 use penumbra_crypto::FullViewingKey;
+use penumbra_custody::policy::{AuthPolicy, PreAuthorizationPolicy};
+use penumbra_custody::soft_kms::{self, SoftKms};
 use penumbra_proto::client::v1alpha1::oblivious_query_service_client::ObliviousQueryServiceClient;
 use penumbra_proto::client::v1alpha1::ChainParametersRequest;
+use penumbra_proto::custody::v1alpha1::custody_protocol_service_server::CustodyProtocolServiceServer;
 use penumbra_proto::view::v1alpha1::view_protocol_service_server::ViewProtocolServiceServer;
 use penumbra_view::ViewService;
-use std::env;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
 use std::str::FromStr;
+use std::{env, fs};
 use tonic::transport::Server;
+#[serde_as]
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ClientConfig {
+    /// Optional KMS config for custody mode
+    pub kms_config: Option<soft_kms::Config>,
+    /// FVK for both view and custody modes
+    pub fvk: FullViewingKey,
+}
```

Comment:
This should be expanded to move most of the current options out of the command-line flags and into the config. We can change the top-level `Opt` structure to take a `--home` argument (similar to `pd` and `tendermint` do) that sets the home directory with all of the configs and data, and load the other options (like the node URL, the host address, etc) from the config structure.

Reply:
We should do this before merging, so we're not rearranging the binary's interface.